connection.go

```diff
@@ -1302,15 +1302,20 @@ func (conn *Connection) ConfiguredTimeout() time.Duration {
 	return conn.opts.Timeout
 }
 
-// OverrideSchema sets Schema for the connection.
-func (conn *Connection) OverrideSchema(s *Schema) {
-	if s != nil {
-		conn.mutex.Lock()
-		defer conn.mutex.Unlock()
-		conn.lockShards()
-		defer conn.unlockShards()
+// SetSchema sets Schema for the connection.
+func (conn *Connection) SetSchema(s Schema) {
```

Comment:
We need to deep copy this fields to make it safe:

https://github.com/tarantool/go-tarantool/blob/954c5b566f6131474eb793333e658197fe328b94/schema.go#L59-L62

Reply:
`Spaces` and `SpacesById` fields of the `Schema` struct store spaces by value.
`Fields` and `FieldsById` fields of the `Space` struct store fields by value. `Index` and `IndexById` fields of the `Space` struct store indexes by value.
`Fields` field of the `Index` struct store `IndexField` by value.

So we dont need to deep copy anything now.

Reply:
> So we dont need to deep copy anything now.

It's not actually true. Values in inner maps/slices (pointer types) could be changes from the external code.

Reply:
Yes, you are right:
`the contents of a slice argument can be modified by a function` from the https://go.dev/blog/slices
`Map types are reference types, like pointers or slices` from the https://go.dev/blog/slices

Anyways, fixed this issue and added a test to check if everything is ok now.